_studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp

```diff
@@ -0,0 +1,1196 @@
+
+    extBuf.RefIdx = (mfxU8)par.film_grain_params_ref_idx;
+
+    extBuf.NumYPoints = (mfxU8)par.num_y_points;
+    for (int i = 0; i < UMC_AV1_DECODER::MAX_POINTS_IN_SCALING_FUNCTION_LUMA; i++)
+    {
+        extBuf.PointY[i].Value = (mfxU8)par.point_y_value[i];
+        extBuf.PointY[i].Scaling = (mfxU8)par.point_y_scaling[i];
+    }
+
+    if (par.chroma_scaling_from_luma)
+        extBuf.Flags |= MFX_FILM_GRAIN_CHROMA_SCALING_FROM_LUMA;
+
+    extBuf.NumCbPoints = (mfxU8)par.num_cb_points;
+    extBuf.NumCrPoints = (mfxU8)par.num_cr_points;
+    for (int i = 0; i < UMC_AV1_DECODER::MAX_POINTS_IN_SCALING_FUNCTION_CHROMA; i++)
+    {
+        extBuf.PointCb[i].Value = (mfxU8)par.point_cb_value[i];
+        extBuf.PointCb[i].Scaling = (mfxU8)par.point_cb_scaling[i];
+        extBuf.PointCr[i].Value = (mfxU8)par.point_cr_value[i];
+        extBuf.PointCr[i].Scaling = (mfxU8)par.point_cr_scaling[i];
+    }
+
+    extBuf.GrainScalingMinus8 = (mfxU8)par.grain_scaling - 8;
+    extBuf.ArCoeffLag = (mfxU8)par.ar_coeff_lag;
+
+    for (int i = 0; i < UMC_AV1_DECODER::MAX_AUTOREG_COEFFS_LUMA; i++)
+        extBuf.ArCoeffsYPlus128[i] = (mfxU8)(par.ar_coeffs_y[i] + 128);
+
+    for (int i = 0; i < UMC_AV1_DECODER::MAX_AUTOREG_COEFFS_CHROMA; i++)
+    {
+        extBuf.ArCoeffsCbPlus128[i] = (mfxU8)(par.ar_coeffs_cb[i] + 128);
+        extBuf.ArCoeffsCrPlus128[i] = (mfxU8)(par.ar_coeffs_cr[i] + 128);
+    }
+
+    extBuf.ArCoeffShiftMinus6 = (mfxU8)par.ar_coeff_shift - 6;
+    extBuf.GrainScaleShift = (mfxU8)par.grain_scale_shift;
+    extBuf.CbMult = (mfxU8)par.cb_mult;
+    extBuf.CbLumaMult = (mfxU8)par.cb_luma_mult;
+    extBuf.CbOffset = (mfxU16)par.cb_offset;
+    extBuf.CrMult = (mfxU8)par.cr_mult;
+    extBuf.CrLumaMult = (mfxU8)par.cr_luma_mult;
+    extBuf.CrOffset = (mfxU16)par.cr_offset;
+
+    if (par.overlap_flag)
+        extBuf.Flags |= MFX_FILM_GRAIN_OVERLAP;
+
+    if (par.clip_to_restricted_range)
+        extBuf.Flags |= MFX_FILM_GRAIN_CLIP_TO_RESTRICTED_RANGE;
+}
+
+mfxStatus VideoDECODEAV1::FillOutputSurface(mfxFrameSurface1** surf_out, mfxFrameSurface1* surface_work, UMC_AV1_DECODER::AV1DecoderFrame* pFrame)
+{
+    MFX_CHECK_NULL_PTR3(surface_work, pFrame, surf_out);
+
+    UMC::FrameData const* fd = pFrame->GetFrameData();
+    MFX_CHECK(fd, MFX_ERR_DEVICE_FAILED);
+
+    mfxVideoParam vp;
+    *surf_out = m_allocator->GetSurface(fd->GetFrameMID(), surface_work, &vp);
+    if (m_opaque)
+       *surf_out = m_core->GetOpaqSurface((*surf_out)->Data.MemId);
+
+    mfxFrameSurface1* surface_out = *surf_out;
+    MFX_CHECK(surface_out, MFX_ERR_INVALID_HANDLE);
+
+    SetFrameType(*pFrame, *surface_out);
+
+    surface_out->Info.FrameId.TemporalId = 0;
+
+    UMC::VideoDataInfo const* vi = fd->GetInfo();
+    MFX_CHECK(vi, MFX_ERR_DEVICE_FAILED);
+
+    surface_out->Info.CropW = static_cast<mfxU16>(pFrame->GetUpscaledWidth());
+    surface_out->Info.CropH = static_cast<mfxU16>(pFrame->GetHeight());
```

Comment:
Where are timestamp, aspect ratio, FrameRateExt, FrameOrder, DataFlag, mfxExtDecodedFrameInfo, etc handling?

Reply:
I will check this with @stasefremov and @IlyaSergeyev.

Reply:
We plan to fix it later.

Reply:
To be tracked in #2231